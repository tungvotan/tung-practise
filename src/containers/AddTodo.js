// @flow

import React, { Component } from 'react'
import { connect } from 'react-redux'
import { addTodo } from '../actions'
type State = {
  value: string
}
class AddTodo extends Component<*, State> {
  
  state = {
    value: ''
  }

  handleSubmit = e => {
    e.preventDefault()
    const value = this.state.value
    if (value) {
      if (!value.trim()) {
        return
      }
      
      this.props.addTodo(value)
      this.setState({
        value: ''
      })
    }
  }

  handleInputChange = (e) => {
    this.setState({
      value: e.target.value
    })
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <input value={this.state.value} onChange={this.handleInputChange} />
          <button type="submit">
            Add Todo
          </button>
        </form>
      </div>
    )
  }  
} 

const mapDispatchToProps = (dispatch) => ({
  addTodo: (value) => dispatch(addTodo(value))
})

export default connect(null, mapDispatchToProps)(AddTodo)
