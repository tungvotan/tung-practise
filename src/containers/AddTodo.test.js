import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AddTodo from './AddTodo';
import configureStore from 'redux-mock-store';

const setup = (setupProps: {}) => {
  const store = configureStore()({ todos: [] });
  const wrapper = shallow(<AddTodo store={store} />);

  return {
    store,
    wrapper
  };
}

describe('AddTodo test', () => {
  test('renders without crashing', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('handle Submit', () => {
    const { wrapper, store } = setup();
    const instance = wrapper.dive().instance()
    let preventDefault = jest.fn();
    instance.state.value = 'Test add todo'
    instance.handleSubmit({ preventDefault });
    expect(store.getActions()).toEqual([
      { id: 0, text: 'Test add todo', type: 'ADD_TODO' }
    ]);
    expect(instance.state.value).toEqual('')
  })
  
  test('handleInputChange', () => {
    const { wrapper } = setup();
    const instance = wrapper.dive().instance()
    instance.handleInputChange({ target:{ value: 'handle change' } })
    expect(instance.state.value).toEqual('handle change')
  })
})