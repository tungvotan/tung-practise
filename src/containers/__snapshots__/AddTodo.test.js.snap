// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo test renders without crashing 1`] = `
<AddTodo
  addTodo={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
